### PR TITLE
[OIS] Open IoT SDK platform environment 

### PR DESCRIPTION
**Pull request by ATmobica**
Provide a new ARM-software platform for the Matter project.
This platform is based on [Open IoT SDK](https://gitlab.arm.com/iot/open-iot-sdk). Open IoT SDK is a reference implementation of [Open-CMSIS-CDI](https://www.open-cmsis-cdi.org/) which defines a common device interface for microcontroller-based devices used in the Internet of Things. It is delivered as a framework of software components with a set of feature-rich example applications.

This initialized PR provides the development environment for the OIS platform.
It contains:
 - Adding OIS Docker image and integration with Vscode environment.
 - Adding OIS sources as submodules.
 - Adding required Python packages to the pigweed environment.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "build": {
         "dockerfile": "Dockerfile",
         "args": {
-            "BUILD_VERSION": "0.6.02"
+            "BUILD_VERSION": "0.6.03"
```
Comment (andy31415) [inline .devcontainer/devcontainer.json:20]: version updates should be done in a separate PR from the dockerfile update: dockerfile takes a while to update and if we commit both at the same time, the devcontainer would be broken for a while

File: integrations/docker/images/chip-build/Dockerfile [hunk 2]
```diff
@@ -60,6 +62,7 @@ RUN set -x \
     shellcheck \
     strace \
     systemd \
+    telnet \
```
[the remark this answers is not in the document]
Comment (ATmobica) [reply integrations/docker/images/chip-build/Dockerfile:65]: These packages are required to run and test OIS examples. It's something specific to the OIS port but I put it in chip-build dockerfile to visible them in the Vscode environment.

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (ATmobica) [reply .devcontainer/devcontainer.json]: Removed from this PR

File: integrations/docker/images/chip-build/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (pan-) [reply integrations/docker/images/chip-build/Dockerfile:65]: The main platform we are integrating is the FVP of the [Corstone-300](https://developer.arm.com/downloads/-/arm-ecosystem-fvps) which is a simulator. Serial ports of the simulator are accessed with telnet.

[the remark this answers is not in the document]
Comment (pan-) [reply integrations/docker/images/chip-build/Dockerfile:65]: In our [fork](https://github.com/ARM-software/connectedhomeip), you can see the complete port. This is the first PR of that upstreaming. 

As you can see in our [GH action](https://github.com/ARM-software/connectedhomeip/actions/runs/3160831183/jobs/5145735906), we do an integration test of the  lock application with commissioning made between the device controller and the ARM application running in the simulator. 

The simulator can be debugged using GDB which facilitates debugging of issues that arise specifically on _real_ embedded targets. We have multiple plans for the future. For example, adding secure storage, secure FW update with TF-M or add Bluetooth and WiFi simulation to the system.|

We would also like to run the certification test suite on the port; in CI.

**Review by pan- — APPROVED** [review]

File: integrations/docker/images/chip-build/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (andy31415) [reply integrations/docker/images/chip-build/Dockerfile:65]: So why in chip-build and not in the platform-specific dockerfile?

[the remark this answers is not in the document]
Comment (andy31415) [reply integrations/docker/images/chip-build/Dockerfile:65]: For VSCode environment, we have chip-build-vscode
If you place it in chip-build, then all platforms inherit it, so you are saying 'telnet is needed by things like nrf, espressif, efr, ....'.

[the remark this answers is not in the document]
Comment (ATmobica) [reply integrations/docker/images/chip-build/Dockerfile]: Right, now the approach is clear to me. I moved these to chip-build-openiotsdk and chip-build-vscode Docker files

**Review by woody-apple — APPROVED** [review]

**Comment by ATmobica** [comment]
@andy31415 @woody-apple Can we go forward with this PR if there are not any more comments?

**Review by andy31415 — APPROVED** [review]
